assets/js/components/user-input/UserInputQuestionnaire.js

```diff
@@ -81,7 +85,11 @@ export default function UserInputQuestionnaire() {
 		setActiveSlug( steps[ activeSlugIndex + 1 ] );
 	}, [ activeSlugIndex ] );
 
-	const goTo = useCallback( ( num = 1 ) => {
+	const goTo = useCallback( ( num = 1, shouldSetSingle = false ) => {
+		// Add the single edit context to the query string when navigating to a specific question on the user-input page.
+		if ( shouldSetSingle ) {
+			setSingle( 'user-input' );
+		}
```

Comment:
@benbowler this functionality doesn't seem to be needed in the questionary component. Let's add a new callback in the preview component that will set this and call the `goTo` function passed to the component via props.

Also, please, use the `useQueryArg` hook in the preview component to get the setter function instead of passing it via props.